spark/src/test/java/org/apache/iceberg/spark/source/TestNullableParquet.java

```diff
@@ -0,0 +1,133 @@
+              true,
+              org.apache.spark.sql.types.Metadata.empty()),
+          new org.apache.spark.sql.types.StructField(
+              "requiredField",
+              org.apache.spark.sql.types.DataTypes.StringType,
+              false,
+              org.apache.spark.sql.types.Metadata.empty())
+      });
+
+  private Schema icebergSchema = new Schema(
+      Types.NestedField.optional(1, "optionalField", Types.StringType.get()),
+      Types.NestedField.required(2, "requiredField", Types.StringType.get()));
+
+  private List<String> data0 = Arrays.asList(
+      "{\"optionalField\": \"a1\", \"requiredField\": \"bid_001\"}",
+      "{\"optionalField\": \"a2\", \"requiredField\": \"bid_002\"}");
+  private List<String> data1 = Arrays.asList(
+      "{\"optionalField\": \"d1\", \"requiredField\": \"bid_101\"}",
+      "{\"optionalField\": \"d2\", \"requiredField\": \"bid_102\"}",
+      "{\"optionalField\": \"d3\", \"requiredField\": \"bid_103\"}",
+      "{\"optionalField\": \"d4\", \"requiredField\": \"bid_104\"}");
+
+  @Test
+  public void testNullablePoC() throws IOException {
+
+    // setup
+    File location = new File(temp.newFolder("parquet"), "test");
+    sourcePath = String.format("%s/nullable_poc/sourceFolder/", location.toString());
+    targetPath = String.format("%s/nullable_poc/targetFolder/", location.toString());
+    tableProperties = Collections.singletonMap(TableProperties.WRITE_NEW_DATA_LOCATION, targetPath);
+
+    sparkSession = SparkSession.builder()
+        .master("local[2]")
+        .appName("NullablePoC")
+        .getOrCreate();
+
+    // create base parquet
+    // we will read this and append to iceberg dataset
+    sparkSession
+        .read().schema(sparkSchema).json(
+        JavaSparkContext.fromSparkContext(sparkSession.sparkContext()).parallelize(data1))
+        .write().parquet(sourcePath);
+
+    // create base iceberg table
+    // this our iceberg dataset to which we will append data
+    new HadoopTables(sparkSession.sparkContext().hadoopConfiguration())
+        .create(
+            icebergSchema,
+            PartitionSpec.builderFor(icebergSchema).identity("requiredField").build(),
+            tableProperties,
+            targetPath);
+
+    // write base iceberg data
+    // this is the initial data inside the iceberg dataset
+    sparkSession
+        .read().schema(sparkSchema).json(
+        JavaSparkContext.fromSparkContext(sparkSession.sparkContext()).parallelize(data0))
+        .write().format("iceberg").mode(SaveMode.Append).save(targetPath);
+
+    // read from parquet and append to iceberg
+    // this ends with:
+    //
+    //     Cannot write incompatible dataset to table with schema
+    //     table {
+    //       1: optionalField: optional string
+    //       2: requiredField: required string
+    //     }
+    //     Problems:
+    //     * requiredField should be required, but is optional
+    //
```

Comment:
should we keep these comments? Looks like description from the problem statement.

Reply:
👍 I will remove the comments.